sqlite: Adding v3.50.2 and v3.50.4

caseyalananderson opened the pull request:
adding sqlite versions 3.46.1 to 3.50.4

Commit:
adding sqlite versions 3.46.1 to 3.50.4

alalazo reviewed (changes requested):
Pipelines are failing, the recipe needs updates

Commit:
reverting to adding only the patched (3.50.2) and most recent (3.50.4) sqlite version and updating the recipe to remove dynamic extensions for recent versions

caseyalananderson committed:
Merge branch 'develop' into add_versions_sqlite

caseyalananderson committed:
Merge branch 'develop' into add_versions_sqlite

caseyalananderson commented:
> Pipelines are failing, the recipe needs updates

Updated recipe to work for 3.50.2 and 3.50.4. Removed all other packages to keep the lastest (3.50.4) and the one with a significant vulnerability fix (3.50.2)

Commit:
fixing dynamic extensions on sqlite

caseyalananderson committed:
Merge branch 'develop' into add_versions_sqlite

Commit:
Re-test commit

Commit:
Merge branch 'add_versions_sqlite' of https://github.com/caseyalananderson/spack-packages into add_versions_sqlite

caseyalananderson commented:
@alalazo recipe updated for 3.50.2 and 3.50.4 fix

caseyalananderson reviewed (commented):
removed all but the latest version and resolved the build issue with the new sqlite releases

caseyalananderson commented:
@rbberger

rbberger commented on repos/spack_repo/builtin/packages/sqlite/package.py:
```diff
@@ -263,7 +267,8 @@ def configure_args(self):
         # Ref: https://www.sqlite.org/loadext.html
-        args.extend(self.enable_or_disable("dynamic-extensions", variant="dynamic_extensions"))
+        if self.spec.satisfies("@:3.48"):
+            args.extend(self.enable_or_disable("dynamic-extensions", variant="dynamic_extensions"))
```
I would make the `dynamic_extensions` variant a conditional.

```python
        variant(
            "dynamic_extensions",
            default=True,
            description="Support loadable extensions",
            when=f"@:3.48 platform={plat}",
        )
```

rbberger replied:
Note, because of the implementation of `enable_or_disable`, once you have a conditional variant you don't need to do a `if` check here anymore, since it will return an empty argument list if the condition is not met.

Commit:
setting sqlite dynamic extensions version requirement as a variant

rbberger commented:
@alalazo any other concerns?
